src/store.js

```diff
@@ -0,0 +1,28 @@
+import Immutable from 'immutable';
+import { createStore, applyMiddleware } from 'redux';
+import thunkMiddleware from 'redux-thunk';
+import createLogger from 'redux-logger';
+import rootReducer from './reducers';
+
+const loggerMiddleware = createLogger({
+  stateTransformer: (state) => {
+    const newState = {};
+
+    for (const i of Object.keys(state)) {
+      if (Immutable.Iterable.isIterable(state[i])) {
+        newState[i] = state[i].toJS();
+      } else {
+        newState[i] = state[i];
+      }
+    }
+
+    return newState;
+  }
+});
+
+const store = createStore(
+  rootReducer,
+  applyMiddleware(thunkMiddleware, loggerMiddleware)
```

Comment:
we should make the module environment aware. May I suggest using Webpack [`definePlugin`](https://github.com/webpack/docs/wiki/list-of-plugins#defineplugin). This is also needed for react. It will log an error in the console if it were to be build with the `-p` flag by webpack.

Therefore we should start with `development` and `production` env. And only expose the redux logger in `development` mode.